pkgs/build-support/testers/testEqualArrayOrMap/assert-equal-array.sh

```diff
@@ -0,0 +1,71 @@
+# shellcheck shell=bash
+
+# Tests if an array is declared.
+isDeclaredArray() {
+  # shellcheck disable=SC2034
+  local -nr arrayRef="$1" && [[ ${!arrayRef@a} =~ a ]]
```

Comment:
Relevant docs to the shell parameter expansion: https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html